### PR TITLE
Allow deleting datasets on disk

### LINKED ISSUE #4628: enable deleting datasets from UI
users that have no access to the filesystem but only use the upload dataset UI have no possibility to remove datasets again. it would be beneficial if they could hide them via an archive button (then the dataset list could filter for those as well)

### PR DESCRIPTION
**Pull request by fm3**
Adds new datastore route `DELETE      /datasets/:organizationName/:dataSetName/deleteOnDisk` (use with care)
Adds a tab to dataset edit view, where this route can be triggered

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- backend: test route `DELETE      /datasets/:organizationName/:dataSetName/deleteOnDisk` (with admin token)
- frontend: check out new view, hit the button, dataset should be moved to .trash directory within organization datasets directory. Should be one after refresh

### Open Questions:
 - how to hide this for non-admins? (how to pass user to view as prop?)
 - how to trigger dataset list refresh on dashboard after redirect

### Issues:
- fixes #4628 

------
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [x] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

File: frontend/javascripts/dashboard/dataset/import_delete_component.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    setIsDeleting(true);
+    await deleteDatasetOnDisk(dataSet.dataStore.url, datasetId);
+    Toast.success(
+      messages["dataset.delete_success"]({
+        datasetName: dataSet.name,
+      }),
+    );
+    await Utils.sleep(2000);
+    setIsDeleting(false);
+    location.href = "/dashboard";
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/dashboard/dataset/import_delete_component.js:48]: We usually use the `RouterHistory` to redirect manually:
here is an [example](https://github.com/scalableminds/webknossos/blob/4fe8507278a037281a10c396fe33ab0dbce33753/frontend/javascripts/admin/auth/login_view.js#L22)

Could you please use it here too?

File: frontend/javascripts/dashboard/dataset/import_delete_component.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import Toast from "libs/toast";
+import messages from "messages";
+
+import { confirmAsync } from "./helper_components";
+
+type Props = {
+  datasetId: APIDatasetId,
+};
+
+export default function ImportSharingComponent({ datasetId }: Props) {
```
[the remark this answers is not in the document]
Comment (MichaelBuessemeyer) [reply frontend/javascripts/dashboard/dataset/import_delete_component.js]: done

File: frontend/javascripts/dashboard/dataset/import_delete_component.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  useEffect(() => {
+    fetch();
+  }, []);
+
+  async function handleDeleteButtonClicked(): Promise<void> {
+    if (!dataSet) {
+      return;
+    }
+    await confirmAsync({
```
[the remark this answers is not in the document]
Comment (MichaelBuessemeyer) [reply frontend/javascripts/dashboard/dataset/import_delete_component.js]: done

**Comment by fm3** [comment]
Thanks a lot for looking at this and even implementing the suggestions! 🎉 
Yes, the backend will only notice the now-missing dataset upon the next inbox check (runs once per minute).
The Refresh button above the dataset list would also cause an inbox check, leading to the list being refreshed. Maybe we can trigger this action also during the deletion before redirecting.
But even then, the front-end shows a cached version of the dataset list until the next refresh, compare https://github.com/scalableminds/webknossos/pull/4640
Do you have any ideas how we could solve this for the deletion?

**Comment by daniel-wer** [comment]
@fm3 Yes, it should be possible to trigger a refresh before redirecting and there should also be a way to show the newest set of datasets, because that's what happens when the "Refresh" button is clicked by the user. I'll have a look at the code :)

**Comment by daniel-wer** [comment]
@fm3 The dataset list should show updated results after deleting a dataset now :)

Regarding the admin-only ToDo - Who should actually have access to the Import/Edit view at all? At a quick glance I didn't see any restrictions (apart from that the user needs access to the dataset).

**Comment by fm3** [comment]
dataset managers should see the links pointing there for all datasets; team managers should see them for the datasets their teams have access to, admins should see them for all datasets. only admins should be allowed to delete

**Comment by MichaelBuessemeyer** [comment]
ToDos for me:
 > dataset managers should see the links pointing there for all datasets; team managers should see them for the datasets their teams have access to, admins should see them for all datasets. I think this is already the case. 


- [ ] check this

> only admins should be allowed to delete


- [ ] implement this

**Comment by MichaelBuessemeyer** [comment]
> @fm3 The dataset list should show updated results after deleting a dataset now :)


works perfectly :ok_hand:

**Comment by MichaelBuessemeyer** [comment]
@daniel-wer Could you please check my latest frontend changes? 

After this, the PR should be ready to go :ship:

**Comment by MichaelBuessemeyer** [comment]
I just had a look at the description.

> - [ ] Updated [documentation](../blob/master/docs) if applicable
> - [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes

@fm3 is updating wk-connect still a thing to do? I'll have a look at the docs

**Comment by daniel-wer** [comment]
Frontend changes look good, thanks for taking this over :+1:

**Comment by fm3** [comment]
I added it to wk-connect’s issue tracker at https://github.com/scalableminds/webknossos-connect/issues/22 
I don’t think it has to be implemented there right away.

**Pull request merged**